app/services/pii/cacher.rb

```diff
@@ -68,9 +68,9 @@ def stale_compound_pii_signature?(profile)
       return false unless profile
       decrypted_pii = fetch
       return false unless decrypted_pii
-      fingerprint = Profile.build_compound_pii_fingerprint(decrypted_pii)
-      return false unless fingerprint
-      Pii::Fingerprinter.stale?(fingerprint, profile.name_zip_birth_year_signature)
```

Comment:
This function is the culprit. The stale check should receive decrypted data so that it can encrypt that and compare the resulting signature to the signature we have in the database. We were passing the signature for both arguments.